### PR TITLE
Update API documentation

### PR DESCRIPTION
**Pull request by sethvargo**
This moves the API docs out of docs and into their own API section. It also adopts the API format currently used by Vault and Consul.

As an added bonus, you also get early beta access to the client-side redirect functionality!

### CODE REVIEW

File: website/redirects.txt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#
```
Comment (sethvargo) [inline website/redirects.txt]: 👀 here - this is important

File: website/source/api/agent.html.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+| Method | Path                         | Produces                   |
+| ------ | ---------------------------- | -------------------------- |
+| `POST` | `/agent/force-leave`         | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes), and
+[required ACLs](/api/index.html#acls).
+
+| Blocking Queries | Consistency Modes | ACL Required |
```
Comment (dadgar) [inline website/source/api/agent.html.md]: I am not sure `Consistency Modes` make sense. There aren't any queries that support default and not stale. The fact that it is either all or none is really getting at the fact that queries either support forwarding to the leader or they are agent local. So it maybe better to have a boolean `Leader Forwarding`

Comment (sethvargo) [inline website/source/api/agent.html.md]: Hey @dadgar 

Thank you for the comment. I was aiming for consistency between Nomad and Consul's docs as much as possible. Do you think having this section at all makes sense for Nomad?

File: website/source/api/agent.html.md [hunk 3]
```diff
@@ -26,9 +26,9 @@ The table below shows this endpoint's support for
 [consistency modes](/api/index.html#consistency-modes), and
```
Comment (dadgar) [inline website/source/api/agent.html.md]: Probably remove `consistency modes` here too

**Pull request merged**